specification/dsdl/dsdl.tex

```diff
@@ -132,30 +132,102 @@ \section{Syntax}\label{sec:dsdl_syntax}
+The legal constant literals can be found in Section \ref{sec:dsdl_primitive_constant_data_types},
+and a comprehensive table of legal operations and their precedence is given below.
+Infix notation is used, meaning a constant evaluation is written as \texttt{A <op> B} for 
+binary operators and \texttt{<op> A} for unary operators, where \texttt{A} and \texttt{B} are constant expressions.
+
+\begin{UAVCANSimpleTable}{Operators for constant expressions}{|c c c c c l|} \label{table:dsdl_const_operators}
+Precedence & Op & A & B & Res & Description \\
+1 & \texttt{!}    &  \texttt{const bool}   & N/A                  & \texttt{const bool}    & Logical not \\
+
+
+2 & \texttt{*}    & \texttt{const int}    & \texttt{const int}   & \texttt{const int}     & Integer multiplication \\
+2 & \texttt{/}    & \texttt{const int}    & \texttt{const int}   & \texttt{const int}     & Integer division \\
+2 & \texttt{\%}   & \texttt{const int}    & \texttt{const int}   & \texttt{const int}     & Integer remainder \\
+
+2 & \texttt{*}    & \texttt{const float}  & \texttt{const float} & \texttt{const float}   & IEE754 multiplication \\
+2 & \texttt{/}    & \texttt{const float}  & \texttt{const float} & \texttt{const float}   & IEE754 division \\
+
+2 & \texttt{\%} & \texttt{\{const int\}}  & \texttt{const int}   & \texttt{\{const int\}} & Element-wise int remainder\\
+
+
+3 & \texttt{+}    & \texttt{const float}  & \texttt{const float} & \texttt{const float}   & IEEE754 sum \\
+3 & \texttt{-}    & \texttt{const float}  & \texttt{const float} & \texttt{const float}   & IEEE754 difference \\
+
+3 & \texttt{+}    & \texttt{const int}    & \texttt{const int}   & \texttt{const int}     & Integer sum \\
+3 & \texttt{-}    & \texttt{const int}    & \texttt{const int}   & \texttt{const int}     & Integer difference \\
+
+
+4 & \texttt{==}   & \texttt{const int}    & \texttt{const int}   & \texttt{const bool}    & Integer equality \\
+4 & \texttt{!=}   & \texttt{const int}    & \texttt{const int}   & \texttt{const bool}    & Integer inequality \\
+4 & \texttt{==}   & \texttt{const bool}   & \texttt{const bool}  & \texttt{const bool}    & Boolean equality \\
+4 & \texttt{!=}   & \texttt{const bool}   & \texttt{const bool}  & \texttt{const bool}    & Boolean inequality \\
+4 & \texttt{==}   & \texttt{const char}   & \texttt{const char}  & \texttt{const bool}    & Character equality \\
+4 & \texttt{!=}   & \texttt{const char}   & \texttt{const char}  & \texttt{const bool}    & Character inequality \\
+4 & \texttt{==}   & \texttt{\{T\}}        & \texttt{\{T\}}       & \texttt{const bool}    & Set equality \\
+4 & \texttt{!=}   & \texttt{\{T\}}        & \texttt{\{T\}}       & \texttt{const bool}    & Set inequality \\
+
+
+5 & \texttt{\&\&} & \texttt{const bool}   & \texttt{const bool}  & \texttt{const bool}    & Logical and \\
+5 & \texttt{\&\&} & \texttt{const bool}   & \texttt{const bool}  & \texttt{const bool}    & Logical or
```

Comment:
I suggest dropping the word `const` from the table because it is redundant. This is not executable code, so, fundamentally, all values we're dealing with are evaluated at the time of parsing, and so all values can be considered constant.